### PR TITLE
NIFI-1919 Add replaceFirst method to expression language

### PR DESCRIPTION
**Pull request by alopresto**
I will open an additional Jira for 1.0 to consider renaming the `replace` method to `replaceAllLiteral` or something similar, and add logic to `StandardFlowSynchronizer` to perform method replacement if this occurs. 

I think there is an opportunity to refactor and consolidate the `*Evaluator` classes -- `ReplaceEvaluator`, `ReplaceFirstEvaluator`, and `ReplaceAllEvaluator` are identical save for the delegate `String` method that is invoked internally.

### CODE REVIEW

**Comment by alopresto** [comment]
Will rebase the merge after it's been reviewed.

**Comment by alopresto** [comment]
Fixed the re-organization of import sections by the IDE.

File: nifi-docs/src/main/asciidoc/expression-language-guide.adoc [hunk 1]
```diff
@@ -884,7 +916,7 @@ Expressions will provide the following results:
 
 
 
-.replaceAll Examples
+.ReplaceAll Examples
```
Comment (mattyb149) [inline nifi-docs/src/main/asciidoc/expression-language-guide.adoc]: Shouldn't this remain lowercase? Also is it necessary for the markdown table or should it be removed to be consistent with other examples sections?

File: nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/functions/ReplaceEvaluator.java [hunk 2]
```diff
@@ -17,7 +17,6 @@
 package org.apache.nifi.attribute.expression.language.evaluation.functions;
 
 import java.util.Map;
-
```
Comment (mattyb149) [inline nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/functions/ReplaceEvaluator.java]: nit pick for unnecessary whitespace change with no other changes in the file

File: nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/test/groovy/org/apache/nifi/processors/script/ExecuteScriptGroovyTest.groovy [hunk 3]
```diff
@@ -149,6 +150,7 @@ class ExecuteScriptGroovyTest extends BaseScriptTest {
         }
     }
 
+    @Ignore("This test fails intermittently when the serial execution happens faster than pooled")
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/test/groovy/org/apache/nifi/processors/script/ExecuteScriptGroovyTest.groovy]: This file should not be listed in the diff. Can you rebase against the latest master? I suspect the SHA of the commit changed from the one on your branch when I did a sign-off or a commit --amend to add the "This closes" line, or something.

**Comment by joewitt** [comment]
Please just make sure we don't create problems for old flows.  Can change names if we have a smooth transition for existing EL statements.

File: nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/functions/ReplaceEvaluator.java (hunk 2 above)
Comment (alopresto) [inline nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/functions/ReplaceEvaluator.java]: Thanks, will remove.

File: nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/test/groovy/org/apache/nifi/processors/script/ExecuteScriptGroovyTest.groovy (hunk 3 above)
Comment (alopresto) [inline nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/test/groovy/org/apache/nifi/processors/script/ExecuteScriptGroovyTest.groovy]: You are probably correct. Have rebased a couple times before now but will figure out why this is here.

**Comment by alopresto** [comment]
@joewitt No existing methods are changing in this fix. A new ticket is filed for NIFI-1938 to address any breaking changes.

**Comment by mattyb149** [comment]
I built and ran various tests with the new (and existing) replaceXYZ functions, everything looks good.

+1
